File: uva_11309.py
def check(num):
    if num<10:
        return 0
    elif num<100:
        if int(num/10)==num%10:
            return 0
        else:
            return 1
    elif num<1000:
        if int(num/100)==num%10:
            return 0
        else:
            return 1
    else:
        h1, h2=int(num/1000), int(num/100)%10
        m=num%100
        if h2*10+h1==m:
            return 0
        else:
            return 1
num=int(input())
while num:
    num-=1
    data=input()
    h=int(data[0:2])
    m=int(data[3:])
    while True:
        m+=1
        if m==60:
            m, h=0, h+1
        if h==24:
            h=0
        if not check(h*100+m):
            break
    print('{:0>2}:{:0>2}'.format(h, m))

File: uva_11309.cpp
# include <iostream>
# include <iomanip>
# include <string>
using namespace std;
bool check(int num){
    if(num<10)
        return 0;
    else if (num<100){
        if (num/10==num%10)
            return 0;
    }
    else if (num <1000){
        if (num/100==num%10)
            return 0;
    }
    else{
        int h1=num/1000, h2=(num/100)%10;
        int m=num%100;
        if (h2*10+h1==m)
            return 0;
        else
            return 1;
    }
    return 1;
}
int main(void){
    int num;
    cin >> num;
    while(num--){
        string data;
        int h, m;
        cin >> data;
        h=(data[0]-'0')*10+(data[1]-'0');
        m=(data[3]-'0')*10+(data[4]-'0');
        do{
            m+=1;
            if (m==60)
                m=0, h+=1;
            if (h==24)
                h=0;
        }while(check(h*100+m));
        cout << setw(2) << setfill('0') << h << ':' 
        << setw(2) << setfill('0') << m << endl;
    }
    return 0;
}